src/ParatextPluginBackTranslationHelper/BackTranslationHelperForm.cs

```diff
@@ -338,30 +505,53 @@ void IBackTranslationHelperDataSource.MoveToNext()
             _setSyncReferenceGroup(_verseReference);
         }
 
-        void IBackTranslationHelperDataSource.WriteToTarget(string text)
+        private bool AreWeChangingTheTarget { get; set; }
+
+        bool IBackTranslationHelperDataSource.WriteToTarget(string text)
         {
             try
             {
+                System.Diagnostics.Debug.WriteLine("PtxBTH: In WriteToTarget");
+
                 _buttonPressed = ButtonPressed.WriteToTarget;
 
                 var vrefTokensTarget = CalculateTargetTokens(_verseReference, _versesReference, text, UsfmTokensSource, UsfmTokensTarget);
                 if (vrefTokensTarget == null)
                 {
-                    RequeryWarning(_verseReference);
-                    return; // nothing to do
+                    // if we don't already have it... it's probably because something was changed in the project
+                    //  external to this form (we lose the data if we get activated).
+                    System.Diagnostics.Debug.WriteLine($"PtxBTH: RequeryWarning");
```

Comment:
did you mean to include _verseReference in this $"{string}"?

Reply:
added both (the single version based on the up-down control) and the range if the actual verses are combined:
                    System.Diagnostics.Debug.WriteLine($"PtxBTH: RequeryWarning: _verseReference: {_verseReference}, _versesReference: {_versesReference}");